federator/src/lib/TransactionSender.js

```diff
@@ -37,11 +37,18 @@ module.exports = class TransactionSender {
         return this.getEthGasPrice();
     }
 
-    async getGasLimit() {
+    async getGasLimit(rawTx) {
+        const estimatedGas = await this.client.eth.estimateGas({
+            gasPrice: rawTx.gasPrice,
+            value: rawTx.value,
+            to: rawTx.to,
+            data: rawTx.data,
+            from: rawTx.from
+        });
         // Gas estimation does not work correctly on RSK and after the London harfork, neither is working on Ethereum
         // example https://etherscan.io/tx/0xd30d6cf428606e2ef3667427b9b6baecb2f4c9cbb44a0c82c735a238ec8f72fb
         // To fix it, we decided to use a hardcoded gas estimation
-        return ESTIMATED_GAS;
+        return +estimatedGas < ESTIMATED_GAS ? ESTIMATED_GAS : +estimatedGas;
```

Comment:
is the `+` needed in the last usage of `estimatedGas`? I mean I know it's necessary at first for the comparison to work, but maybe the default value can just be `estimatedGas` instead of `+estimatedGas`?

Reply:
If i recall correctly default value is a BigNumber so i rather cast it to Number to safetly compare it with ESTIMATED_GAS